### PR TITLE
Xontrib

### PR DESCRIPTION
**Pull request by scopatz**
This is an initial attempt at having extensions.  I still need to add docs, but I figured I would throw it out there before I went to sleep tonight

### CODE REVIEW

**Comment by scopatz** [comment]
Ok everyone, I have added documentation to the xontrib branch.  I feel like it is ready for a look. I would be happy for it to go in as is. More complex features can come later if they are needed, though I am not convinced that they are at this point.  In any event, let me know what you think!

File: xonsh/xonfig.py [hunk 1]
```diff
@@ -82,6 +83,24 @@
...
+WIZARD_XONTRIB = """
+{hr}
+
+                           {{BOLD_WHITE}}Xontribs{{NO_COLOR}}
+                           {{YELLOW}}--------{{NO_COLOR}}
+No shell is complete without extentions, and xonsh is no exception. Xonsh
+extensions are called {{BOLD_GREEN}}xontribs{{NO_COLOR}}, or xonsh contributions.
+Xontribs are dynamically loadable, either by importing them directly or by
+using the 'xontrib' command. However, you can also configure xonsh to load
+xontribs autoumatically on startup prior to loading the run control files.
```
Comment (melund) [inline xonsh/xonfig.py]: autoumatically -> automatically

File: xonsh/xonfig.py [hunk 2]
```diff
@@ -82,6 +83,24 @@
...
+This allows the xontrib to be used immeadiately in your xonshrc files.
```
Comment (melund) [inline xonsh/xonfig.py]: immeadiately -> immediately

File: xonsh/xonfig.py [hunk 3]
```diff
@@ -82,6 +83,24 @@
 
 WIZARD_ENV_QUESTION = "Would you like to set env vars now, " + YN
 
+WIZARD_XONTRIB = """
+{hr}
+
+                           {{BOLD_WHITE}}Xontribs{{NO_COLOR}}
+                           {{YELLOW}}--------{{NO_COLOR}}
+No shell is complete without extentions, and xonsh is no exception. Xonsh
```
Comment (melund) [inline xonsh/xonfig.py]: extentions -> extensions

File: docs/tutorial_xontrib.rst [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+.. code-block:: python
+
+    import xontrib.mpl
+    from xontrib import myext
+    from mypkg.show import *
+
+
+Listing Known Xontribs
+======================
+In addition to loading extensions, the ``xontrib`` command also allow you to
```
Comment (melund) [inline docs/tutorial_xontrib.rst]: allow -> allows

**Comment by melund** [comment]
Nice. It will be interesting to see what people come up with. Almost ready for PyCon?

**Comment by melund** [comment]
LGTM

**Comment by scopatz** [comment]
Thanks for the review @melund! I have fixed up all of the issues you mentioned.

> Nice. It will be interesting to see what people come up with. 

Agreed!

> Almost ready for PyCon?

I am in the process of getting ready to get ready, which means that I am behind as usual :)

**Comment by scopatz** [comment]
Also like you said on gitter @melund, I am in favor of keeping this as simple as possible for the moment, util we find out what people really need / want. So in light of that, I think that this should be "feature complete" for now and able to be merged in.

**Pull request merged**